streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -174,6 +199,32 @@ public void process(final Record<KIn, VIn> record) {
                     keyClass,
                     valueClass),
                 e);
+        } catch (final FailedProcessingException | TaskCorruptedException e) {
```

Comment:
Are you sure, you do not also need to consider `TaskMigratedException` here? I think, you need. Could you please also check other exceptions?

Reply:
I test is needed that verifies whether the `TaskCorruptedException` etc. are ignored by the handler.

Reply:
Exceptions with a special treatment in the `StreamThread` have been ignored: https://github.com/apache/kafka/blob/dceda5e647cf44226c2be5e148e0be60ce665db9/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java#L722

`TaskMigratedException` (added to ignore) and `TaskCorruptedException` have been ignored. I believe users won't expect these exceptions to be handled, that's why they're just rethrown as is.

`UnsupportedVersionException` should not need to be ignored as it cannot be thrown during process